v0/destinations/clickup/util.js

```diff
@@ -0,0 +1,372 @@
+      return getDropDown(customField, value);
+    case "labels":
+      return getLabels(customField, value);
+    case "location":
+      return getLocation(value);
+    default:
+      return value;
+  }
+};
+
+/**
+ * Function to to build the custom field.
+ * returns the array of custom field with each object contains id,value
+ * @param {*} keyToCustomFieldName [{"from":"url","to":"Url"}]
+ * @param {*} properties {"url":"https://www.rudderstack.com/"}
+ * @param {*} listId
+ * @param {*} apiToken
+ * @returns [{"id":"b0f40a94-ea2a-4998-a514-8074d0eddcde","value":"https://www.rudderstack.com/"}]
+ */
+const customFieldsBuilder = async (
+  keyToCustomFieldName,
+  properties,
+  listId,
+  apiToken
+) => {
+  const responseArray = [];
+  // retrieve available clickup custom field for the given list
+  const availableCustomFields = await retrieveCustomFields(listId, apiToken);
+  // convert array to hashMap with key as field name and value as custom field object
+  const availableCustomFieldsMap = getHashFromArrayWithValueAsObject(
+    availableCustomFields,
+    "name",
+    false
+  );
+
+  // convert destination.Config.keyToCustomFieldName to hashMap
+  const configCustomFieldsMap = getHashFromArray(
+    keyToCustomFieldName,
+    "from",
+    "to",
+    false
+  );
+
+  Object.keys(configCustomFieldsMap).forEach(propertiesKey => {
+    let fieldValue = properties[propertiesKey];
+    if (fieldValue) {
+      const fieldName = configCustomFieldsMap[propertiesKey];
+      const customField = availableCustomFieldsMap[fieldName];
+      validateCustomField(customField, fieldValue);
+      fieldValue = findCustomFieldValue(customField, fieldValue);
+      if (fieldValue && customField) {
+        responseArray.push({
+          id: customField.id,
+          value: fieldValue
+        });
+      }
+    }
+  });
+
+  return responseArray;
+};
+
+/**
+ * Function to discard any event that is not a part of destination.Config.whitelistedEvents
+ * @param {*} message
+ * @param {*} destination
+ */
+const eventFiltering = (message, destination) => {
+  const { whitelistedEvents } = destination.Config;
+  const { event } = message;
+  const nonEmptyWhitelistedEvents = whitelistedEvents?.filter(
+    whiteListedEvent => whiteListedEvent.eventName
+  );
+
+  if (nonEmptyWhitelistedEvents && nonEmptyWhitelistedEvents.length > 0) {
+    const allowEvent = nonEmptyWhitelistedEvents.some(
+      whiteListedEvent =>
+        whiteListedEvent.eventName.toLowerCase() === event.toLowerCase()
+    );
+    if (!allowEvent) {
+      throw new CustomError(
+        "Event Discarded. To allow this event, add this in Allowlist",
+        400
+      );
+    }
+  }
+};
+
+const isDefinedAndNotNullAndNotEmpty = value => {
+  return !(
+    value === undefined ||
+    value === null ||
+    Number.isNaN(value) ||
+    (typeof value === "object" && Object.keys(value).length === 0) ||
+    (typeof value === "string" && value.trim().length === 0)
+  );
+};
```

Comment:
This is already in our util function. Use that.

Reply:
There is an issue with global util function.
isDefinedAndNotNullAndNotEmpty is returning true for Integer, boolean

Reply:
if the value is supposed to be not defined, there should not be any dependability on the type of the variable. Is there any unique scenario that is happening?

Reply:
yes, 
If i am using global utils isDefinedAndNotNullAndNotEmpty then from the output, 
following fields are omitted.
{
  "timeEstimate": 10800000,
  "priority": 1,
  "includeDueDateTime": true,
  "includeStartDateTime": true
}